docs/source/main_classes/deepspeed.mdx

```diff
@@ -1751,6 +1751,171 @@ Please note that if you're not using the [`Trainer`] integration, you're complet
 [[autodoc]] deepspeed.HfDeepSpeedConfig
     - all
 
+### DeepSpeed ZeRO Inference
+
+Here is an example of how one could do DeepSpeed ZeRO Inference without using [`Trainer`] when one can't fit a model onto a single GPU. The solution includes using additional GPUs or/and offloading GPU memory to CPU memory.
+
+The important nuance to understand here is that the way ZeRO is designed you can process different inputs on different gpus in parallel.
+
+The example has copious notes and is self-documenting.
```

Comment:
```suggestion
The example has copious notes and is self-documented.
```

Reply:
self-documenting is an idiom.
https://en.wikipedia.org/wiki/Self-documenting_code

I have never seen the "self-documented" form.

Reply:
Oh ok.